frontend/packages/dev-console/src/components/pipelines/hooks.ts

```diff
@@ -0,0 +1,31 @@
+import { match as RMatch } from 'react-router-dom';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useSelector } from 'react-redux';
+import { getBreadcrumbPath } from '@console/internal/components/utils/breadcrumbs';
+import { K8sKind } from '@console/internal/module/k8s';
+import { RootState } from '@console/internal/redux';
+import { getActivePerspective } from '@console/internal/reducers/ui';
+import { pipelinesTab } from '../../utils/pipeline-utils';
+
+export const usePipelinesBreadcrumbsFor = (
+  kindObj: K8sKind,
+  match: RMatch<any>,
+  namespace: string,
+  navOption: string,
+) => {
+  const activePerspective = useSelector((state: RootState) => getActivePerspective(state));
+  return [
+    {
+      name: kindObj.labelPlural,
+      path:
+        activePerspective === 'admin'
+          ? `/${navOption}/ns/${namespace}${
```

Comment:
You have a bug here if you're in "all-namespaces" when you go into the details page - leaving via the breadcrumb causes a not found issue with `#ALL_NS#`.